sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts

```diff
@@ -143,8 +143,15 @@ export class CallArgSerializer {
   async newObjectArg(objectId: string): Promise<ObjectArg> {
     const object = await this.provider.getObject(objectId);
     const initialSharedVersion = getSharedObjectInitialVersion(object);
+    
+    const mutable = true; // Defaulted to True to match current behavior.
+    const api = await this.provider.getRpcApiVersion();
+
     if (initialSharedVersion) {
-      return { Shared: { objectId, initialSharedVersion } };
+      const object_args = api?.major === 0 && api?.minor < 24 
```

Comment:
```suggestion
      const object_args = api?.major === 0 && api?.minor < 25 
```

We should update this to 25